### PR TITLE
Feat/serverless post

### PR DESCRIPTION
**Pull request by yjose**
* serverless part 1

### CODE REVIEW

File: blog/2019-07-28/index.md [hunk 1]
```diff
@@ -0,0 +1,113 @@
+---
+date: 2019-07-28
+title: "The power of Serverless : Part 1"
+categories: ["react", "serverless"]
+published: true
+keywords: ["react", "serverless", "netlify", "auth0", "FaunaDB"]
+banner: "./banner.png"
+---
+
+Nowadays, for sure you have heard of Serverless and how this new hype can help you build application fast and almost for free for small apps which the case for most side project you are trying to build.
```
Comment (smakosh) [inline blog/2019-07-28/index.md]: applications* or "an application"

File: blog/2019-07-28/index.md [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+
+In this article I will share with you a step-by-step from search to launch guide to how you can build complex logic apps using Serverless services and your brain.
```
Comment (smakosh) [inline blog/2019-07-28/index.md]: ...step-by-step guide, from search to launch... sounds better!

File: blog/2019-07-28/index.md [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+
+<br />
+
+> Our use case today is building a new feature on Geek Blabla website.
+
+### About Geeksblabla
+
+[GeeksBlabla](https://geeksblabla.com/) is one of the [DevC Casablanca](https://www.facebook.com/groups/DevC.Casablanca/) community products. its a 3 to 5 people show talking about a specific dev topic or interviewing a success Moroccan person in IT. Briefly it's a Geeks blabla.
+
+At the moment I am writing this article there are more than 30 hours of great and valuable discussions. make sure to take a look.
```
Comment (smakosh) [inline blog/2019-07-28/index.md]: At the moment I am writing this article, there are more than....

File: blog/2019-07-28/index.md [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+
+> Before we start, we need to answer some questions to make sure we understand what we are trying to build.
+
+#### What we are trying to build?
```
Comment (smakosh) [inline blog/2019-07-28/index.md]: What are we trying to build?*

File: blog/2019-07-28/index.md [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+
+One of the most requested features is to add a new suggest page where people can suggest a new topic and also votes for existed ones. And of course with some requirements:
```
Comment (smakosh) [inline blog/2019-07-28/index.md]: .... is adding a new page where.... and also vote for existing ones and of course with few requirements:*

File: blog/2019-07-28/index.md [hunk 6]
```diff
@@ -0,0 +1,113 @@
...
+
+- The Process should be fast and secure.
+- Prevent spam submit.
```
Comment (smakosh) [inline blog/2019-07-28/index.md]: submissions*

File: blog/2019-07-28/index.md [hunk 7]
```diff
@@ -0,0 +1,113 @@
...
+- The User only allowed to vote one time per suggestion.
```
Comment (smakosh) [inline blog/2019-07-28/index.md]: The user is only allowed to vote...

File: blog/2019-07-28/index.md [hunk 8]
```diff
@@ -0,0 +1,113 @@
...
+
+#### What we have?
```
Comment (smakosh) [inline blog/2019-07-28/index.md]: What do we have?*

File: blog/2019-07-28/index.md [hunk 9]
```diff
@@ -0,0 +1,113 @@
...
+- How to build a voting system?
+- Voting system platform? ( Existing sass product)
+- Best Serverless databases?
+- Serverless resources?
+
+> I already have some answers in my mind but having new ideas would definitely help. ( make it a habit)
+
+**How to build a voting system?**
+
+Guess what! this is a wrong question in our case, all Google result is related to blockchain and mathematics theories.
```
Comment (smakosh) [inline blog/2019-07-28/index.md]: 🤣🤣 🤣🤣 🤣🤣

File: blog/2019-07-28/index.md [hunk 10]
```diff
@@ -0,0 +1,113 @@
...
+
+**Voting system platform? ( Existing sass product**)
+
+Recently, you can find a sass project for every feature you want: contact page, authentication notification almost everything. that’s why I thought we can find a sass voting system with a simple API to use. Unfortunately, I only found some complex platform and API that would be impossible to be part of our existing stack.
```
Comment (smakosh) [inline blog/2019-07-28/index.md]: that's why I thought we could find*

File: blog/2019-07-28/index.md [hunk 11]
```diff
@@ -0,0 +1,113 @@
...
+
+After a while, I remember a page I already see in [expo](https://expo.canny.io/feature-requests) website which let developers suggest new features to be added on the expo platform and Guess what this is perfectly what we are trying to build.
```
Comment (smakosh) [inline blog/2019-07-28/index.md]: I recall instead of I remember a page

Comment (smakosh) [reply blog/2019-07-28/index.md]: I've already seen*

File: blog/2019-07-28/index.md [hunk 12]
```diff
@@ -0,0 +1,113 @@
...
+
+As you can see you can suggest new features and other developers can votes.
+
+![](https://cdn-images-1.medium.com/max/800/0*ZE8tM1GKYi7GilHc)
+
+First look, I noticed “ power by canny ” link. Ok let’s see what is Canny
+
+Reading the features page and feeling happy look like this is the best choice for us. But unfortunately the pricing page is not what we want to hear, no freemium plan and starter plan starts from \$50/mo.
```
Comment (smakosh) [inline blog/2019-07-28/index.md]: looks like*
is not what we want to hear => isn't what we're anticipating

File: blog/2019-07-28/index.md [hunk 13]
```diff
@@ -0,0 +1,113 @@
...
+
+![](https://cdn-images-1.medium.com/max/800/0*Yp4JdRhzCW6LCGuZ)
+
+> Good try but at least we can get inspired from how it works.
+
+**Best Serverless resources?**
+
+Jumping from page to page, I found a great website in CSS-tricks team that needs to be added in your bookmarks.
+[The power of Serverless for front-end developers](https://serverless.css-tricks.com/) is a complete list of serverless services and resources organized by sections in one place, Almost any think you will need in your project : data storage, cloud functions, payments, etc.
```
Comment (smakosh) [inline blog/2019-07-28/index.md]: Almost any think => almost anything

File: blog/2019-07-28/index.md [hunk 14]
```diff
@@ -0,0 +1,113 @@
...
+
+### Conclusion
+
+Taking some time to search for the best resource in our case and this is My final notes based on the first step ( Search):
+
+- Authentication is the best way to guarantee one vote per user. There are some other approaches but they are not secure.
+- [**Auth0**](https://auth0.com/) for authentication: Free for 7000 active users which is enough for us.
```
Comment (smakosh) [inline blog/2019-07-28/index.md]: Keep in mind that Auth0 is taking all that data and you might want to let your users know about that.

File: blog/2019-07-28/index.md [hunk 15]
```diff
@@ -0,0 +1,113 @@
...
+- Using [**Netlify**](https://www.netlify.com/products/) functions to write our API: I love Netlify and we already using it to host the static website.
```
Comment (smakosh) [inline blog/2019-07-28/index.md]: I love Netlify and we're already using it...

File: blog/2019-07-28/index.md (hunk 14 above)
Comment (yjose) [reply blog/2019-07-28/index.md]: yeah correct, we only request public data

**Pull request merged**